### PR TITLE
New recipe for Nuitka project

### PR DESCRIPTION
**Pull request by frol**
What is [Nuitka](http://nuitka.net/pages/overview.html)?

Nuitka is a Python compiler.

It's fully compatible with Python 2.6, 2.7, 3.2, 3.3, 3.4, and 3.5.

You feed it your Python app, it does a lot of clever things, and spits out an executable or extension module.

Free license (Apache).

### CODE REVIEW

File: recipes/nuitka/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+$PYTHON setup.py install
```
Comment (ocefpaf) [inline recipes/nuitka/build.sh]: You can remove the `build.sh` file (but keep the `bld.bat`) and add a `build/script` in the metatadata. See below.

File: recipes/nuitka/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  name: nuitka
+  version: {{ version }}
+
+source:
+  fn: Nuitka-{{ version }}.tar.gz
+  url: http://nuitka.net/releases/Nuitka-{{ version }}.tar.gz
+  md5: 95508d2293ddb0cedd7bfd504e1b27e1
+
+build:
+  number: 0
```
Comment (ocefpaf) [inline recipes/nuitka/meta.yaml]: Add a `script: python setup.py install` here.

**Comment by ocefpaf** [comment]
Awesome recipe! I made minor suggestion, but we can merge it as soon as you are ready.

**Comment by frol** [comment]
@ocefpaf I have applied your suggestions and squashed it in one commit.

**Comment by ocefpaf** [comment]
Thanks! As soon as we have green again I'll merge.

**Comment by jakirkham** [comment]
@kayhayen, we are adding `nuitka` to [conda-forge]( https://conda-forge.github.io/ ). This is the community packaging system for use with [`conda`]( http://conda.pydata.org/docs/ ). Packages built here are built with CI and deployed to the [conda-forge channel]( https://anaconda.org/conda-forge/ ) automatically after committing on `master` of a feedstock (a repo with the recipe plus automatically maintained CI scripts). Eventually, packages submitted here will be included in the `defaults` channel after some auditing process as [mentioned]( https://github.com/conda-forge/conda-forge.github.io/issues/22#issuecomment-197117472 ). If there is anything here that you see is off or needs to be fixed, please let us know. If you would like to have access to the [nuitka feedstock/recipe]( http://github.com/conda-forge/nuitka-feedstock/ ), I would happily add you to the maintainers list so you can have commit privileges for this recipe. This would be useful for updating the version for instance. Thanks again for such a nifty Python implementation.

**Comment by kayhayen** [comment]
Hello there,

thanks for coming about to me with this. I think I can contribute to the
recipe. The version can be derived from "bin/nuitka --version", if that's
feasible. Otherwise I could simply automated bumping it to what is there.
Do you think the CI can also do tests, e.g. "./tests/run-tests
--skip-reflection-test". I don't have any MacOS test coverage, and I love
to gain it this way. And I would appreciate a "nuitka-unstable" recipe so
as finde breakage in the pre-release that I make after everything passed on
my Windows/Linux buildbot tests.

Let me know if that are acceptable directions. Otherwise, glad to be a part
of Anaconda, I will point it out on the download page, once I had a chance
to convince myself that it works nicely.

Yours,
Kay

2016-04-05 17:23 GMT+02:00 jakirkham <notifications@github.com>:

**Comment by frol** [comment]
Pinging @ocefpaf @jakirkham.

While I am the author of the recipe, and I know that running tests on a build stage is completely possible, I don't know if it won't be an abuse of the Conda-Forge CI servers. I can help @kayhayen with adding tests execution to the current recipe and with adding an unstable recipe.

P.S. Adding @kayhayen to the [nuitka-feedstock](https://github.com/conda-forge/nuitka-feedstock) might be also a good idea. Can I do this by simply adding his nickname to the `extra/recipe-maintainers` in the `meta.yaml`?

**Comment by ocefpaf** [comment]
@frol running the full test suit is something that we are still discussion. You already have a grasp of the problem: CIs time vs More testing.

Here is the rule of thumb. If the tests are light weight and won't blow up the CIs. Go for it. If you are OK with a simple "my package works" kind of test, that is good enough for us.

PS: Feel free to move this discussion to https://github.com/conda-forge/nuitka-feedstock now.

**Comment by ocefpaf** [comment]
>  Can I do this by simply adding his nickname to the extra/recipe-maintainers in the meta.yaml?

Yes. Just send a PR to https://github.com/conda-forge/nuitka-feedstock and, once merged, he will be added automatically as a maintainer.

**Pull request merged**